pkg/templates/nstemplatetiers/nstemplatetier_generator_whitebox_test.go

```diff
@@ -460,130 +505,29 @@ func TestNewTierTemplate(t *testing.T) {
 }
 
 func assertClusterResourcesTemplate(t *testing.T, decoder runtime.Decoder, actual templatev1.Template, tier string) {
-	switch tier {
-	case "base", "test":
+	if !basedOnOtherTier(tier) {
 		expected := templatev1.Template{}
 		content, err := Asset(fmt.Sprintf("%s/cluster.yaml", tier))
 		require.NoError(t, err)
 		_, _, err = decoder.Decode(content, nil, &expected)
 		require.NoError(t, err)
 		assert.Equal(t, expected, actual)
```

Comment:
How about a small sanity check that the template contains some objects?
```suggestion
		assert.Equal(t, expected, actual)
		assert.NotEmpty(t, actual.Objects)
```